packages/expo-updates/ios/EXUpdates/Update/EXUpdatesLegacyUpdate.m

```diff
@@ -46,18 +46,13 @@ + (EXUpdatesUpdate *)updateWithLegacyManifest:(EXUpdatesLegacyRawManifest *)mani
 
   NSString *bundleUrlString = manifest.bundleUrl;
   NSArray *assets = manifest.bundledAssets ?: @[];
-
-  id runtimeVersion = manifest.runtimeVersion;
-  if (runtimeVersion && [runtimeVersion isKindOfClass:[NSDictionary class]]) {
-    id runtimeVersionIos = ((NSDictionary *)runtimeVersion)[@"ios"];
-    NSAssert([runtimeVersionIos isKindOfClass:[NSString class]], @"runtimeVersion['ios'] should be a string");
-    update.runtimeVersion = (NSString *)runtimeVersionIos;
-  } else if (runtimeVersion && [runtimeVersion isKindOfClass:[NSString class]]) {
-    update.runtimeVersion = (NSString *)runtimeVersion;
+  
+  // Use the manifest's runtimeVersion if it is not null AND the config's runtimeVersion is not null
+  if (manifest.runtimeVersion && config.runtimeVersion) {
```

Comment:
```suggestion
  if (manifest.runtimeVersion != nil && config.runtimeVersion != nil) {
```

imo it's better to be explicit that we're checking that the value is nil, rather than just some falsey value. this also matches our behvior from android side